x-pack/plugin/security/src/internalClusterTest/java/org/elasticsearch/xpack/security/authc/ApiKeyIntegTests.java

```diff
@@ -1092,7 +1098,13 @@ public void testGetAllApiKeys() throws InterruptedException, ExecutionException
         List<CreateApiKeyResponse> userWithManageOwnApiKeyRoleApiKeys = userWithManageOwnTuple.v1();
 
         final Client client = client().filterWithHeader(
-            Collections.singletonMap("Authorization", basicAuthHeaderValue("user_with_manage_api_key_role", TEST_PASSWORD_SECURE_STRING))
+            Collections.singletonMap(
+                "Authorization",
+                basicAuthHeaderValue(
+                    randomFrom("user_with_read_security_role", "user_with_manage_api_key_role"),
+                    TEST_PASSWORD_SECURE_STRING
+                )
+            )
```

Comment:
Nit: We could also randomise the client in the following helper method to cover both Get and Query API keys:

https://github.com/elastic/elasticsearch/blob/0302d2db2de0434c567e94dd369ca922708af896/x-pack/plugin/security/src/internalClusterTest/java/org/elasticsearch/xpack/security/authc/ApiKeyIntegTests.java#L2571

Personally I am happy with just unit tests for the privileges. So this really is just a Nit. Feel free to skip.

Reply:
I have pushed 0d32fa3ce49